todo_app/app.py

```diff
@@ -25,25 +30,73 @@ def create_app():
             format="%(asctime)s %(levelname)s %(name)s %(threadName)s: %(message)s",
         )
 
+    app_user = User()
+
+    # OAuth Login
+    login_manager = LoginManager()
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        # logic to redirect to GH auth flow if unauthed
+
+        client_id: str = os.environ.get("GH_CLIENTID")
+        callback_uri = os.environ.get("CALLBACK_URI")
+
+        return redirect(
+            f"https://github.com/login/oauth/authorize?client_id={client_id}&redirect_uri={callback_uri}"
+        )
+
+    @login_manager.user_loader
+    def load_user(user_id: str) -> User:
+        app_user = User()
```

Comment:
It feels very suspicious that this doesn't use the `user_id` - is there a reason that's not passed through to the User object?

Reply:
Given you use the DB elsewhere; you'll probably need to look at using the id to look up the user in the DB rather than just creating the user directly